packages/dashboard-frontend/src/components/EditorTools/index.tsx

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2018-2021 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import { Button, Divider, Flex, FlexItem } from '@patternfly/react-core';
+import { CompressIcon, CopyIcon, DownloadIcon, ExpandIcon } from '@patternfly/react-icons';
+import React from 'react';
+import CopyToClipboard from 'react-copy-to-clipboard';
+import stringify from '../../services/helpers/editor';
+import styles from './index.module.css';
+import devfileApi from '../../services/devfileApi';
+
+type Props = {
+  content: che.Workspace | devfileApi.DevWorkspace | che.WorkspaceDevfile | devfileApi.Devfile;
+  handleExpand: (isExpand: boolean) => void;
+};
+
+type State = {
+  copied?: boolean;
+  isExpanded: boolean;
+  contentText: string;
+  isDevWorkspace: boolean;
+  contentBlobUrl: string;
+};
+
+class EditorTools extends React.PureComponent<Props, State> {
+  private copiedTimer: number | undefined;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      isExpanded: false,
+      contentText: '',
+      isDevWorkspace: false,
+      contentBlobUrl: '',
+    };
+  }
+
+  public componentDidMount(): void {
+    const { content } = this.props;
+    try {
+      const contentText = stringify(content);
+      const isDevWorkspace = this.isDevWorkspace(content);
+      const contentBlobUrl = URL.createObjectURL(
+        new Blob([contentText], { type: 'application/x-yaml' }),
+      );
+      this.setState({ contentText, isDevWorkspace, contentBlobUrl });
+    } catch (e) {
+      console.error(e);
+    }
+  }
+
+  public componentDidUpdate(): void {
+    const { content } = this.props;
+    try {
+      const contentText = stringify(content);
+      if (contentText !== this.state.contentText) {
+        const isDevWorkspace = this.isDevWorkspace(content);
+        const contentBlobUrl = URL.createObjectURL(
+          new Blob([contentText], { type: 'application/x-yaml' }),
+        );
+        this.setState({ contentText, isDevWorkspace, contentBlobUrl });
+      }
+    } catch (e) {
+      console.error(e);
+    }
+  }
+
+  private handleExpand(): void {
+    if (this.state.isExpanded) {
+      window.postMessage('show-navbar', '*');
+      const isExpanded = false;
+      this.setState({ isExpanded });
+      this.props.handleExpand(isExpanded);
+    } else {
+      window.postMessage('hide-navbar', '*');
+      const isExpanded = true;
+      this.setState({ isExpanded });
+      this.props.handleExpand(isExpanded);
+    }
+  }
+
+  private isDevWorkspace(
```

Comment:
The method name is inaccurate since it also returns `true` for Che7 workspaces.

```suggestion
  private isWorkspace(
```